sidekick/views.py

```diff
@@ -1,8 +1,71 @@
+import requests
+import json
 from os import environ
 from django.http import JsonResponse
+from rest_framework import status
+
+from .models import Organization
 
 
 def health(request):
     app_id = environ.get("MARATHON_APP_ID", None)
     ver = environ.get("MARATHON_APP_VERSION", None)
     return JsonResponse({"id": app_id, "version": ver})
+
+
+def send_wa_template_message(request):
+    data = request.GET.dict()
+
+    required_params = [
+        "org_id",
+        "wa_id",
+        "namespace",
+        "element_name",
+        "message",
+    ]
+    for field in required_params:
+        if field not in data:
+            return JsonResponse(
+                {
+                    "error": "Required fields: {}".format(
+                        ", ".join(required_params)
+                    )
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+    org_id = data["org_id"]
+    wa_id = data["wa_id"]
+    namespace = data["namespace"]
+    element_name = data["element_name"]
+    message = data["message"]
+
+    try:
+        org = Organization.objects.get(id=org_id)
+    except Organization.DoesNotExist:
+        return JsonResponse(
+            {"error": "Organization not found"},
+            status=status.HTTP_400_BAD_REQUEST,
+        )
+
+    headers = {
+        "Authorization": "Bearer {}".format(org.engage_token),
+        "Content-Type": "application/json",
+    }
+
+    result = requests.post(
+        "{}v1/messages".format(org.engage_url),
```

Comment:
This assumes that the `engage_url` has a trailing slash. So we either need to handle that here or as part of the `save` function on the model. Or we could use a `URLField`

Reply:
Also, is there some way we can make sure that we're defaulting to `https`?